lib/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.cpp

```diff
@@ -505,9 +505,12 @@ struct TritonGenericPattern : public OpConversionPattern<Op> {
   LogicalResult
   matchAndRewrite(Op op, typename Op::Adaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    Type retType = this->getTypeConverter()->convertType(op.getType());
+    SmallVector<Type, 4> retTypes;
+    for (auto ty : op.getOperation()->getResultTypes()) {
+      retTypes.push_back(this->getTypeConverter()->convertType(ty));
+    }
     addNamedAttrs(
-        rewriter.replaceOpWithNewOp<Op>(op, retType, adaptor.getOperands()),
+        rewriter.replaceOpWithNewOp<Op>(op, retTypes, adaptor.getOperands()),
```

Comment:
This is so the generic pattern works for `AtomicStoreOp` which has no return value.